cmd/zpool/zpool_main.c

```diff
@@ -8439,14 +8450,43 @@ zpool_do_scrub(int argc, char **argv)
 		(void) fprintf(stderr, gettext("invalid option "
 		    "combination :-s and -p are mutually exclusive\n"));
 		usage(B_FALSE);
+	} else if (is_pause && is_txg_continue) {
+		(void) fprintf(stderr, gettext("invalid option "
+		    "combination :-p and -C are mutually exclusive\n"));
+		usage(B_FALSE);
+	} else if (is_stop && is_txg_continue) {
+		(void) fprintf(stderr, gettext("invalid option "
+		    "combination :-s and -C are mutually exclusive\n"));
+		usage(B_FALSE);
+	} else if (is_error_scrub && is_txg_continue) {
+		(void) fprintf(stderr, gettext("invalid option "
+		    "combination :-e and -C are mutually exclusive\n"));
+		usage(B_FALSE);
+	} else if (is_pause && is_recent_scrub) {
+		(void) fprintf(stderr, gettext("invalid option "
+		    "combination :-p and -R are mutually exclusive\n"));
+		usage(B_FALSE);
+	} else if (is_stop && is_recent_scrub) {
+		(void) fprintf(stderr, gettext("invalid option "
+		    "combination :-s and -R are mutually exclusive\n"));
+		usage(B_FALSE);
+	} else if (is_error_scrub && is_recent_scrub) {
+		(void) fprintf(stderr, gettext("invalid option "
+		    "combination :-e and -R are mutually exclusive\n"));
+		usage(B_FALSE);
 	} else {
-		if (is_error_scrub)
+		if (is_error_scrub) {
 			cb.cb_type = POOL_SCAN_ERRORSCRUB;
+		}
 
-		if (is_pause) {
+		if (is_recent_scrub) {
+			cb.cb_scrub_cmd = POOL_SCRUB_RECENT_TXGS;
+		} else if (is_pause) {
 			cb.cb_scrub_cmd = POOL_SCRUB_PAUSE;
 		} else if (is_stop) {
 			cb.cb_type = POOL_SCAN_NONE;
+		} else if (is_txg_continue) {
+			cb.cb_scrub_cmd = POOL_SCRUB_FROM_LAST_TXG;
```

Comment:
It looks like `-C` and `-R` should also be mutually exclusive.

Reply:
Done.

Reply:
I worry what happen when there would be 10 options, not 5. ;)  I'd probably just count the number of options specified and printed universal message if more than one set.  And while there, if nothing else, I think space should be after the semicolon in "combination :",  not before.